flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTypeUtil.java

```diff
@@ -274,6 +274,9 @@ public TypeInfo visit(DateType dateType) {
 
 		@Override
 		public TypeInfo visit(TimestampType timestampType) {
+			if (timestampType.getPrecision() != 9) {
```

Comment:
I'm wondering if this is too restrictive. While Hive only supports a precision of 9, it should be able to handle any ts type with a precision less than 9.  Thus, Hive timestamp needs to be mapped to Flink ts(9) while any ts(<=9) from Flink can be mapped to Hive timestamp. What do you think?

Reply:
Ideally yes, we should support any ts(<=9). But to do that, we need a way to store the precision of timestamp in HMS, and restore the precision when retrieving the table schema. I prefer to leave it to follow up tasks.

Reply:
-1 for this restrictive check.
It is very important that we use this way to support UDF too. If we have this restrictive check here, UDF use timestamp type will be very restrictive.
I prefer to revolve this problem now.

Reply:
The restriction is maintained when creating table in HiveCatalog. Added a switch to turn off the check so that functions calling this util method can support different precisions.